src/Block.jsx

```diff
@@ -0,0 +1,163 @@
+      return (
+        <Card fluid>
+          <Card.Content>
+            <Card.Description>
+              <Icon name='warning circle' style={{height: '100%'}} />
+              block undefined
+            </Card.Description>
+          </Card.Content>
+        </Card>
+      )
+    } else {
+      return (
+            <Card fluid>
+              <Card.Content>
+                <Card.Description>
+                  <List divided verticalAlign='middle'>
+                    {this.props.blockNumber ?  <List.Item>
+                      <List.Content>
+                        BlockNumber
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Rspan>{formatBlockNumber(this.state.block.number)}</Rspan>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {this.props.timestamp ?  <List.Item>
+                      <List.Content>
+                        TimeStamp
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <div style={{fontSize:'x-small'}}>{this.state.block.timestamp.toString()}</div>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {this.props.transactions ?  <List.Item>
+                      <List.Content>
+                        Transactions
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Ra>{this.state.block.transactions.length}</Ra> txns
+                      </List.Content>
+                    </List.Item> : "" }
+                    {this.props.hash ?  <List.Item>
+                      <List.Content>
+                        Hash
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Hash value={this.state.block.hash}></Hash>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {this.props.parentHash ?  <List.Item>
+                      <List.Content>
+                        ParentHash
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Hash value={this.state.block.parentHash}></Hash>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {this.props.sha3Uncles ?  <List.Item>
+                      <List.Content>
+                        Sha3Uncles
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Hash value={this.state.block.sha3Uncles}></Hash>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {(this.props.author || this.props.miner) ?  <List.Item>
+                      <List.Content floated='right'>
+                        <Hash value={this.state.block.author}></Hash>
```

Comment:
Didn't want to double the icon but yes looks better. Will see if the other fields can look as nice too.